tests/Integration/Behaviour/Features/Context/Domain/AttachmentFeatureContext.php

```diff
@@ -0,0 +1,120 @@
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Attachment;
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+use PrestaShopException;
+use RuntimeException;
+
+class AttachmentFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I add new attachment :reference with following properties:
+     *
+     * @param string $reference
+     * @param TableNode $tableNode
+     */
+    public function addAttachment(string $reference, TableNode $tableNode): void
+    {
+        $data = $tableNode->getRowsHash();
+        $fileName = $data['file_name'];
+
+        $destination = $this->uploadDummyFile($fileName);
+
+        $attachment = new Attachment();
+        $attachment->description = $this->parseLocalizedArray($data['description']);
+        $attachment->name = $this->parseLocalizedArray($data['name']);
+        $attachment->file_name = $fileName;
+        $attachment->mime = mime_content_type($destination);
+        $attachment->file = pathinfo($destination, PATHINFO_BASENAME);
+
+        $attachment->add();
+
+        $this->getSharedStorage()->set($reference, (int) $attachment->id);
+    }
+
+    /**
+     * @Then attachment :reference should have following properties:
+     *
+     * @param string $reference
+     * @param TableNode $tableNode
+     */
+    public function assertAttachmentProperties(string $reference, TableNode $tableNode): void
+    {
+        $attachment = $this->getAttachment($reference);
+        $data = $tableNode->getRowsHash();
+
+        Assert::assertEquals($this->parseLocalizedArray($data['description']), $attachment->description);
+        Assert::assertEquals($this->parseLocalizedArray($data['name']), $attachment->name);
+        Assert::assertEquals($data['file_name'], $attachment->file_name);
+        Assert::assertEquals($data['mime'], $attachment->mime);
+        Assert::assertEquals($data['size'], $attachment->file_size);
+    }
+
+    /**
+     * @param string $reference
+     *
+     * @return Attachment
+     *
+     * @throws PrestaShopException
+     */
+    private function getAttachment(string $reference): Attachment
+    {
+        $id = $this->getSharedStorage()->get($reference);
+        $attachment = new Attachment($id);
+
+        if (!$attachment->id) {
+            throw new RuntimeException(sprintf('Failed to load attachment with id %d', $id));
+        }
+
+        return $attachment;
+    }
+
+    /**
+     * @param string $fileName
+     *
+     * @return string uploaded file destination path including the file name
+     */
+    private function uploadDummyFile(string $fileName): string
+    {
+        $source = _PS_ROOT_DIR_ . '/tests/Resources/dummyFile/' . $fileName;
+
+        if (!is_file($source)) {
+            throw new RuntimeException('%s is not a file', $source);
+        }
+
+        $destination = _PS_DOWNLOAD_DIR_ . $fileName;
+        copy($source, $destination);
+        chmod($destination, 7550);
```

Comment:
Why not `6440` ? Does it need to be executable?

Reply:
Thanks for not blocking the PR for this. 
opened another one, that isn't in a hurry & won't block anything
https://github.com/PrestaShop/PrestaShop/pull/21209